### ISSUE TITLE
Public sharing and pretty URL

### ISSUE THREAD

**Issue by fossxplorer** [owncloud/core#1074]
Hi,
it would be benefical for oC to have prettier URL for shared links. Currently it looks like http://example.com/public.php?service=files&t=659082f5a8a1ab14a19ce3fd0cfc6720.
DropBox' equivalent looks like https://www.dropbox.com/s/e12pt647unyo3a8/myfile.ogg.

One suggestion might be to get the URL on the form  http://example.com/public.php/files/t/659082f5a8a1ab14a19ce3fd0cfc6720/myfile.ogg as public.php might be part of the core logic in oC. 
Do also note the file name being displayed at the end of the URL. Please do not encapsulate  the file name in the token as there are cases where this might help, e.g to be able to search among many bookmarked publicly shared links!

I would like to use the opportunity to thank all the devs behind oC!

@butonic @bartv2

**Comment by bkerensa** [comment]
Why did you close this @Kondou-ger your example are not pretty urls at all a pretty url might be something like https://yourcloud.com/s/bjork-mp3s or https://yourcloud.com/s/meeting-minutes2014 or something but https://yourcloud.com/s/tbzID5QMCYhtVOs is absolutely not pretty or easy to remember.

**Comment by Kondou-ger** [comment]
@bkerensa urls like https://yourcloud.com/s/tbzID5QMCYhtVOs/myfile.ogg are being worked on somewhere.

Self defined urls like https://yourcloud.com/s/kondous-pdfs are a different thing, but I don't think we even need them, since you are not supposed to remember these but send them via chat/email/…, and click on them, not type them in.

For someone who isn't tech-savvy typing in the whole yourcloud.com/s/something might already be challenging enough from my experiences.

@butonic @DeepDiver1975 @jancborchardt opinions?

**Comment by jerrac** [comment]
@Kondou-ger 
> since you are not supposed to remember these but send them via chat/email/…, and click on them, not type them in.
I disagree. URLs are also a way to know where you are. If making them readable isn't important, then why do we have DNS? Why not just use ip addresses? 

Here's another use case. If you are hosting a lot of public files that you want search engines to index. Yes, OwnCloud might not be the most efficient method, but if you have it implemented already, why not? For search engines, having a solid pretty URL is important. That's why most Drupal sites use the PathAuto module to change node/number into blog/this-is-my-title.

Or maybe you are printing a url in a snail mail newsletter? site.com/jerrac/file.jpg is far quicker to type than even a url shortened url, since you don't have to keep double checking that you typed the right sequence of letters and numbers.

Basically, there are use cases for pretty urls. I'm fine with closing this topic if the OwnCloud devs think it's not a feature for core OwnCloud. But don't close it as fixed. Fixed would either be it actually getting into OwnCloud, or someone successfully building an app for it.

**Comment by bkerensa** [comment]
Why wouldn't we want to remember a link? Dropbox and Box have offered this for years now (self defined URLs)

Use case: I run a non profits IT and we post all our documents in Owncloud as we have to keep records long term we think link to share links from wordpress to things like minutes and financial records. We keep years of records.

Having to guess whether s3Ks7kAfjsj is the hash URL for our 2013 finance records versus a nice clean self defined named URL is a breathe of fresh air.

Its also the reason we created DNS versus just using IP addresses for websites... People can't remember strings of numbers.

**Comment by bkerensa** [comment]
If we can't seek to be an open source solution that seeks parity with priority cloud platforms then why continue? Even the enterprise edition should have this

**Comment by dahamsta** [comment]
+1 for readable URLs.

**Comment by Kondou-ger** [comment]
@jerrac @bkerensa @dahamsta that's just my opinion. Let's see what other devs say about this.

**Comment by jancborchardt** [comment]
Something like  `myowncloud.com/[some user ID]/that-document.odf` would be great indeed. And yes, that’s how Dropbox URLs look like. I don’t know of the technical or security implications though.

@DeepDiver1975 @LukasReschke your takes please.

**Comment by dahamsta** [comment]
Dropbox uses a userid rather than a username. Ideally I'd like the option of both, with the userid being the default.

https://www.dropbox.com/s/vpqym5e0saf43bt/.nomedia?dl=0

**Comment by LukasReschke** [comment]
> Something like myowncloud.com/[some user ID]/that-document.odf would be great indeed. And yes, that’s how Dropbox URLs look like. I don’t know of the technical or security implications though.

That would be possible. Though the `[some user ID]` is actually a share ID and NOT an user id as this leaks some information you don't necessarily want to leak at this point.

The "Show raw file" feature will not be possible at the moment without having another domain support.

**Comment by jancborchardt** [comment]
@LukasReschke yeah, that sounds fine. The important thing here is that there is _some_ human-readable part in the URL, in this case the filename.

**Comment by jerrac** [comment]
What about user configurable aliases? Say, my user name is blah, but I
configure my share url to be foo/subdir/filenam.ext.

--David Reagan

**Comment by jancborchardt** [comment]
@jerrac as @LukasReschke said, usernames won’t be leaked/used anyway.

**Comment by jerrac** [comment]
> Though the [some user ID] is actually a share ID
So, a share ID isn't a random string of characters? It would be something
human readable?

--David Reagan

**Comment by LukasReschke** [comment]
The Share ID will be random. The filename however the file was called. We will not add a custom URL function at the moment for multiple reasons (including the risk of social engineering etc.)

If you really want URLs that look like you want please use a URL shortener :)

**Comment by bkerensa** [comment]
Clearly other cloud storage platforms dont think the risk of social engineering is high.

**Comment by LukasReschke** [comment]
1. We are not others. We are we ;)
2. Honestly this discussion is useless we can spend time better instead of an endless discussion. If you really want it that way you are free to write an ownCloud app that behaves tjid way.

=> Locked.

**Comment by MorrisJobke** [comment]
Fixed in 9.0 https://github.com/owncloud/core/pull/14081
